### PR TITLE
MesonToolchain improvements

### PR DESCRIPTION
**Pull request by franramirez688**
Docs for https://github.com/conan-io/conan/pull/11632 https://github.com/conan-io/conan/pull/11618

### CODE REVIEW

File: reference/conanfile/tools/meson/mesontoolchain.rst [hunk 1]
```diff
@@ -116,6 +116,50 @@ The ``MesonToolchain`` only works with the ``PkgConfigDeps`` generator.
 Please, do not use other generators, as they can have overlapping definitions that can conflict.
 
 
+Default directories
++++++++++++++++++++++
+
+Since Conan 1.51, ``MesonToolchain`` manages some of the directories used by Meson. They are `conan_meson_*.ini`
```
[the remark this answers is not in the document]
Comment (franramirez688) [reply reference/conanfile/tools/meson/mesontoolchain.rst:122]: ok, let me rephrase it

**Review by SSE4 — APPROVED** [review]

File: reference/conanfile/tools/meson/mesontoolchain.rst [hunk 2]
```diff
@@ -200,3 +244,15 @@ to the ``MesonToolchain`` class interface. For instance:
             tc = MesonToolchain(self)
             tc.cpp = "/path/to/other/compiler"
             tc.generate()
+
+
+Objective-C arguments
+++++++++++++++++++++++
+
+Since Conan 1.51, it's been introduced some specific Objective-C arguments: ``objc``, ``objcpp``, ``objc_args``,
```
Comment (SSE4) [inline reference/conanfile/tools/meson/mesontoolchain.rst:252]:
```suggestion
Since Conan 1.51, it's been introduced some specific Objective-C/Objective-C++ arguments: ``objc``, ``objcpp``, ``objc_args``,
```

File: reference/conanfile/tools/meson/mesontoolchain.rst [hunk 3]
```diff
@@ -116,6 +116,50 @@ The ``MesonToolchain`` only works with the ``PkgConfigDeps`` generator.
...
+
+        def generate(self):
+            tc = MesonToolchain(self)
+            print(tc.project_options["datadir"])  # Will print '["res"]'
+            tc.generate()
+
+
+.. note::
+
+    All of them are saved only if they have any value. If the values is ``None``, they won't be mentioned
```
Comment (SSE4) [inline reference/conanfile/tools/meson/mesontoolchain.rst:159]:
```suggestion
    All of them are saved only if they have any value. If the values are``None``, they won't be mentioned
```

File: reference/conanfile/tools/meson/mesontoolchain.rst [hunk 4]
```diff
@@ -116,6 +116,50 @@ The ``MesonToolchain`` only works with the ``PkgConfigDeps`` generator.
...
+    class App(ConanFile):
+        settings = "os", "arch", "compiler", "build_type"
+
+        def layout(self):
+            self.folders.build = "build"
+            self.cpp.package.resdirs = ["res"]
+
+        def generate(self):
+            tc = MesonToolchain(self)
+            print(tc.project_options["datadir"])  # Will print '["res"]'
```
Comment (memsharded) [inline reference/conanfile/tools/meson/mesontoolchain.rst:153]: In general it is better to use ``self.output.info()`` etc, because this output will be automatically prefixed with the package scope.

**Review by memsharded — APPROVED** [review]

**Review by lasote — APPROVED** [review]
